mlflow/projects/docker.py

```diff
@@ -26,18 +26,23 @@
 
 def validate_docker_installation():
     """
-    Verify if Docker is installed on host machine.
+    Verify if Docker is installed and running on host machine.
     """
-    try:
-        docker_path = "docker"
-        process._exec_cmd([docker_path, "--help"], throw_on_error=False)
-    except EnvironmentError:
+    if shutil.which("docker") is None:
         raise ExecutionException(
             "Could not find Docker executable. "
             "Ensure Docker is installed as per the instructions "
             "at https://docs.docker.com/install/overview/."
         )
 
+    cmd = ["docker", "ps"]
+    prc = process._exec_cmd(cmd, throw_on_error=False)
+    if prc.returncode != 0:
+        joined_cmd = " ".join(cmd)
+        raise ExecutionException(
+            f"Ran `{joined_cmd}` to ensure docker daemon is running but it failed: {prc.stderr}"
```

Comment:
Nit: also log the "prc.stdout" which might be useful for showing what's wrong.

Reply:
@WeichenXu123 Done!